src/main/java/com/google/cloud/tools/managedcloudsdk/internal/download/Downloader.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.internal.download;
+
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+/**
+ * Downloader for downloading a single Cloud SDK archive, it is a callable we use in chained
+ * asynchronous calls.
+ */
+public interface Downloader extends Callable<Path> {
```

Comment:
Interface feels like overkill if there's only one implementation. Have you considered merging this with StandardDownloader?

Reply:
Kind of a style thing I guess, it is consistent with the other code for this feature that will be uploaded.